api/apps/api/src/modules/scenarios/scenarios.service.ts

```diff
@@ -88,13 +90,16 @@ export class ScenariosService {
     const validatedMetadata = this.getPayloadWithValidatedMetadata(input);
     const scenario = await this.crudService.create(validatedMetadata, info);
     await this.planningUnitsLinkerService.link(scenario);
+    await this.planningUnitsStatusCalculatorService.calculatedProtectionStatusForPlanningUnitsIn(scenario);
     return scenario;
   }
 
   async update(scenarioId: string, input: UpdateScenarioDTO) {
     await this.assertScenario(scenarioId);
     const validatedMetadata = this.getPayloadWithValidatedMetadata(input);
-    return this.crudService.update(scenarioId, validatedMetadata);
+    const scenario = await this.crudService.update(scenarioId, validatedMetadata);
+    await this.planningUnitsStatusCalculatorService.calculatedProtectionStatusForPlanningUnitsIn(scenario);
+    return scenario;
```

Comment:
Another thing is that if getting service is ever extended with returning PU as well, it would return invalid (old - before calculations) data.

Reply:
you are right - we can ignore this case right now as it's not happening in practice, but I guess a quick way out would be to keep the update as is now and fetch the scenario again after calculating the default inclusion status.